ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+from .models import OrderedProduct
+from client.models import Client
+from supplier_product.models import SupplierProduct
+from delivery_location.models import DeliveryLocation
+from product.models import Product
+from supplier.models import Supplier
+from django.shortcuts import redirect
+from django.shortcuts import render
+from django.contrib.auth.decorators import user_passes_test
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.models import User
+
+
+def client_check(user):
+    try:
+        return Client.objects.get(client_account=user)
+    except ObjectDoesNotExist:
+        return False
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def change_quantity(request, order_id):
+    try:
+        quantity = int(request.GET['quantity'])
+        ordered_product = OrderedProduct.objects.get(id=int(order_id))
+    except Exception:
+        return redirect('/client/')
+    if (quantity > 0):
+        ordered_product.increase_quantity(quantity)
+    elif (quantity < 0):
+        ordered_product.decrease_quantity(abs(quantity))
+    return redirect('/client/')
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def delete_order(request, order_id):
+    try:
+        OrderedProduct.objects.get(id=int(order_id)).delete_ordered_product()
+    except ObjectDoesNotExist:
+        pass
+    return redirect('/client/')
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def order(request):
+    return render(request, 'ordered_product/new_order.html', {})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def search_product_by_product_name(request):
+    prod_name = str(request.GET['product_name'])
+    message = ""
+    try:
+        product = Product.objects.get(product_name=prod_name)
+        supplier_products = SupplierProduct.objects.filter(qr_code=product)
```

Comment:
We need a test case in which supplier_products contains no products. I don't think this should be in the try and I think you might have a bug here, as it won't show the message you expect.

Reply:
Got it.. I will fix it